deepchecks/nlp/utils/text_properties.py

```diff
@@ -870,40 +925,54 @@ def calculate_builtin_properties(
-            {'Sample': truncate_string(text, max_length=20) if text else 'EMPTY STRING'},
-            refresh=False
-        )
-        if pd.isna(text):
-            for prop in text_properties:
-                calculated_properties[prop['name']].append(np.nan)
-            continue
-        sample_language = run_available_kwargs(language, text=text, **kwargs)
-        if is_language_property_requested:
-            calculated_properties['Language'].append(sample_language)
-        kwargs['language_property_result'] = sample_language  # Pass the language property result to other properties
+    for i in tqdm(range(0, len(raw_text), batch_size)):
+        batch = raw_text[i:i + batch_size]
+        batch_properties = defaultdict(list)
 
+        # filtering out empty sequences
+        nan_indices = {i for i, seq in enumerate(batch) if pd.isna(seq) is True}
+        filtered_sequences = [e for i, e in enumerate(batch) if i not in nan_indices]
+
+        samples_language = _batch_wrapper(text_batch=filtered_sequences, func=language, **kwargs)
+        if is_language_property_requested:
+            batch_properties['Language'].extend(samples_language)
+            calculated_properties['Language'].extend(samples_language)
+        kwargs['language_property_result'] = samples_language  # Pass the language property to other properties
+
+        non_english_indices = set()
+        if ignore_non_english_samples_for_english_properties:
+            non_english_indices = {i for i, (seq, lang) in enumerate(zip(filtered_sequences, samples_language))
+                                   if lang != 'en'}
         for prop in text_properties:
             if prop['name'] in import_warnings:  # Skip properties that failed to import:
-                calculated_properties[prop['name']].append(np.nan)
-            elif sample_language != 'en' and prop['name'] in english_properties_names \
-                    and ignore_non_english_samples_for_english_properties is True:
-                calculated_properties[prop['name']].append(np.nan)
+                batch_properties[prop['name']].extend([np.nan] * len(batch))
             else:
+                if prop['name'] in english_properties_names \
+                        and ignore_non_english_samples_for_english_properties is True:
+                    filtered_sequences = [e for i, e in enumerate(filtered_sequences) if i not in non_english_indices]
+                kwargs['batch_size'] = batch_size
                 try:
-                    value = run_available_kwargs(prop['method'], text=text, **kwargs)
-                    calculated_properties[prop['name']].append(value)
+                    if prop['name'] in BATCH_PROPERTIES:
+                        value = run_available_kwargs(func=prop['method'], text_batch=filtered_sequences, **kwargs)
+                    else:
+                        value = _batch_wrapper(text_batch=filtered_sequences, func=prop['method'], **kwargs)
```

Comment:
I kinda hate this, can we perhaps add _batch_wrapper inside the original property dict? It's ugly duplication but I don't like that we have these gazillion lists of property types...

Reply:
I think it makes sense to have batched & non-batched properties